tests/Functional/Command/Cache/LineItemCacheWarmerCommandTest.php

```diff
@@ -0,0 +1,147 @@
+
+use Doctrine\Common\Cache\CacheProvider;
+use Doctrine\ORM\EntityManagerInterface;
+use LogicException;
+use Monolog\Logger;
+use OAT\SimpleRoster\Command\Cache\LineItemCacheWarmerCommand;
+use OAT\SimpleRoster\Command\Cache\LtiInstanceCacheWarmerCommand;
+use OAT\SimpleRoster\Generator\LineItemCacheIdGenerator;
+use OAT\SimpleRoster\Repository\LineItemRepository;
+use OAT\SimpleRoster\Repository\LtiInstanceRepository;
+use OAT\SimpleRoster\Tests\Traits\CommandDisplayNormalizerTrait;
+use OAT\SimpleRoster\Tests\Traits\DatabaseTestingTrait;
+use OAT\SimpleRoster\Tests\Traits\LoggerTestingTrait;
+use Symfony\Bundle\FrameworkBundle\Console\Application;
+use Symfony\Bundle\FrameworkBundle\Test\KernelTestCase;
+use Symfony\Component\Console\Tester\CommandTester;
+
+class LineItemCacheWarmerCommandTest extends KernelTestCase
+{
+    use DatabaseTestingTrait;
+    use LoggerTestingTrait;
+    use CommandDisplayNormalizerTrait;
+
+    /** @var CommandTester */
+    private $commandTester;
+
+    /** @var CacheProvider */
+    private $resultCache;
+
+    /** @var LineItemCacheIdGenerator */
+    private $lineItemCacheIdGenerator;
+
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $kernel = self::bootKernel();
+
+        $application = new Application($kernel);
+        $this->commandTester = new CommandTester($application->find(LineItemCacheWarmerCommand::NAME));
+
+        /** @var EntityManagerInterface $entityManager */
+        $entityManager = self::$container->get(EntityManagerInterface::class);
+        $resultCacheImplementation = $entityManager->getConfiguration()->getResultCacheImpl();
+
+        if (!$resultCacheImplementation instanceof CacheProvider) {
+            throw new LogicException('Doctrine result cache is not configured.');
+        }
+
+        $this->resultCache = $resultCacheImplementation;
+        $this->lineItemCacheIdGenerator = self::$container->get(LineItemCacheIdGenerator::class);
+
+        $this->setUpDatabase();
+        $this->setUpTestLogHandler('cache_warmup');
+    }
+
+    public function testItEncapsulatesAnyUnexpectedExceptions(): void
+    {
+        $kernel = self::bootKernel();
+
+        $lineItemRepository = $this->createMock(LineItemRepository::class);
+        $lineItemRepository
+            ->method('findAllAsCollection')
+            ->willThrowException(new LogicException('Yaaay'));
+
+        self::$container->set('test.line_item_repository', $lineItemRepository);
+
+        $application = new Application($kernel);
+        $commandTester = new CommandTester($application->find(LineItemCacheWarmerCommand::NAME));
+
+        self::assertSame(1, $commandTester->execute([], ['capture_stderr_separately' => true]));
+        self::assertStringContainsString(
+            '[ERROR] An unexpected error occurred: Yaaay',
+            $commandTester->getDisplay()
+        );
+    }
+
+    public function testItDisplaysWarningIfThereAreNoLtiInstancesIngested(): void
```

Comment:
Copy paste power :) line item, not lti instance

Reply:
Thank you :)